docs/API.md

```diff
@@ -437,8 +411,82 @@ Options
+the value is expressed using an enumeration type (see below).
+
+* `int64_t` - some options need the paramter specified as 64-bit integer
+
+* `bool` - You need to use a boolean type, the builtin one in C++ and the
+one from `<stdbool.h>` for C, although when setting an option, passing the
+value through `int` type is also tolerated and properly recognized. When
+getting an option, however, you better use the `bool` type, although you
+can risk passing a variable of `int` type initialized with 0 and then check if
+the resulting value is equal to 0 (just don't compare the result with 1).
+
+* `string` - When setting, you pass the character array pointer as value
+and the string length as length. When getting, you have to pass a long
+enough array (with specified size in the size variable). Every option with
+this type that can be read should specify the maximum length of that array.
+
+
+Enumeration types used in options
+---------------------------------
+
+
+### 1. `SRT_TRANSTYPE`. Used by `SRTO_TRANSTYPE` option:
+
+* `SRTT_LIVE`: Live mode.
+* `SRTT_FILE`: File mode.
+
+See below [Transmission types](#transmission-types) for details.
+
+
+### 2. `SRT_KM_STATE`. This defined encryption state as performed by the Key
+Material Exchange, used by `SRTO_RCVKMSTATE`, `SRTO_SNDKMSTATE` and
+`SRTO_KMSTATE` options:
+
+- `SRT_KM_S_UNSECURED`: no encryption/descryption. If this state is only on
+the receiver, received encrypted packets will be dropped.
+
+- `SRT_KM_S_SECURING`: pending security (HSv4 only). This is a temporary state
+used only if the connection is made has HSv4 and the Key Material Exchange is
+not finished yet. On HSv5 not possible because Key Material Exchange for the
+initial key is done in the handshake.
+
+- `SRT_KM_S_SECURED`: KM exchange was successful and the data will be sent
+enrypted and decrypted by the receiver. This state is only possible on both
+sides in both directions simultaneously.
+
+- `SRT_KM_S_NOSECRET`: (HSv5 only) This site has set password, but data will 
+be received as plain. This also means that sending the data from this party
+will not be received by the peer due to impossible decryption.
+
+- `SRT_KM_S_BADSECRET`: The password is wrong (set differently on each party),
+encrypted payloads won't be decrypted in any direction.
+
+Note that with the default value of `SRTO_ENFORCEDENCRYPTION` option (true),
+the state is equal on both sides in both directions and it can be only
+`SRT_KM_S_UNSECURED` or `SRT_KM_S_SECURED` (in other cases the connection
+is rejected). Otherwise it may happen that either both sides have different
+password and the state is `SRT_KM_S_BADSECRET` in both direction, or we
+have a situation where only one party has set a password, in which case
+the party that hasn't set the password has:
+
+* `SRT_KM_S_NOSECRET` on `SRTO_RCVKMSTATE`
+* `SRT_KM_S_UNSECURED` on `SRTO_SNDKMSTATE`
+
+and its peer the other way around.
```

Comment:
```suggestion
Note that with the default value of `SRTO_ENFORCEDENCRYPTION` option (true),
the state is equal on both sides in both directions, and it can be only
`SRT_KM_S_UNSECURED` or `SRT_KM_S_SECURED` (in other cases the connection
is rejected). Otherwise it may happen that either both sides have a different
password and the state is `SRT_KM_S_BADSECRET` in both directions, or only one 
party has set a password, in which case the party that hasn't set the password has:

* `SRT_KM_S_NOSECRET` on `SRTO_RCVKMSTATE`
* `SRT_KM_S_UNSECURED` on `SRTO_SNDKMSTATE`

In such a case, its peer would have the inverse settings.
```

Reply:
I changed the last line to `and the peer similarly, with the directions inverted.`, but that also doesn't sound good for me.

What I meant is that peer will have:

* `SRT_KM_S_NOSECRET` on `SRTO_SNDKMSTATE`
* `SRT_KM_S_UNSECURED` on `SRTO_RCVKMSTATE`

Just didn't want to go so elaborate.

Reply:
How about:

```
party has set a password, in which case the KM state settings are as follows:

|                          | `SRTO_RCVKMSTATE`    | `SRTO_SNDKMSTATE`    |
|--------------------------|----------------------|----------------------|
| Party with no password:  | `SRT_KM_S_NOSECRET`  | `SRT_KM_S_UNSECURED` |
| Peer with password:      | `SRT_KM_S_UNSECURED` | `SRT_KM_S_NOSECRET`  |

```
And then just remove the last line.

Reply:
Magnificent.